demo/src/main/java/eu/kevin/demo/main/MainViewModel.kt

```diff
@@ -17,71 +28,170 @@ import kotlinx.coroutines.flow.receiveAsFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-class MainViewModel : ViewModel() {
+internal class MainViewModel constructor(
+    private val getCreditorsUseCase: GetCreditorsUseCase
+) : ViewModel() {
 
-    private val kevinAuthClient = KevinApiClientFactory(
-        baseUrl = BuildConfig.KEVIN_API_URL,
-        "",
-        timeout = 120000,
-        logLevel = LogLevel.NONE
-    ).createClient(null)
+    private val _viewState = MutableStateFlow(MainViewState())
 
-    private val _viewState = MutableStateFlow<MainViewState>(MainViewState.Loading(false))
     private val _viewAction = Channel<MainViewAction>(Channel.BUFFERED)
 
+    private val donationConfiguration = DonationConfiguration()
```

Comment:
doesn't it makes sense to include donationConfiguration into the state itself as it has variables inside which are basically related to the view state?